src/rashdf/geom.py

```diff
@@ -142,13 +142,105 @@ def mesh_cell_faces(self) -> GeoDataFrame:
         return GeoDataFrame(face_dict, geometry="geometry", crs=self.projection())
 
     def bc_lines(self) -> GeoDataFrame:
-        raise NotImplementedError
+        """Return the 2D mesh area boundary condition lines.
+        
+        Returns
+        -------
+        GeoDataFrame
+            A GeoDataFrame containing the 2D mesh area boundary condition lines if they exist.
+        """
+        if "/Geometry/Boundary Condition Lines" not in self:
+            return GeoDataFrame()
+        bc_line_data = self["/Geometry/Boundary Condition Lines"]
+        bc_line_ids = range(bc_line_data["Attributes"][()].shape[0])
+        v_conv_str = np.vectorize(convert_ras_hdf_string)
+        names = v_conv_str(bc_line_data["Attributes"][()]["Name"])
+        mesh_names = v_conv_str(bc_line_data["Attributes"][()]["SA-2D"])
+        types = v_conv_str(bc_line_data["Attributes"][()]["Type"])
+        multi_lines = list()
+        for pnt_start, pnt_cnt, part_start, part_cnt in bc_line_data["Polyline Info"][()]:
+            points = bc_line_data["Polyline Points"][()][pnt_start:pnt_start+pnt_cnt]
+            parts = bc_line_data["Polyline Parts"][()][part_start:part_start+part_cnt]
+            multi_lines.append(
+                MultiLineString(
```

Comment:
Can we make this use `LineString` for single-part polylines and `MultiLineString` for multipart? Would be nice to avoid `MultiLineString`s that are actually single-part.

Reply:
Updated to use `LineString` when single-part.